prusti-specs/src/extern_spec_rewriter.rs

```diff
@@ -201,76 +136,470 @@ fn rewrite_method_inputs(item_ty: &syn::Type, method: &mut ImplItemMethod) ->
             }
         }
         args.push_punct(syn::token::Comma::default());
-    };
+    }
     args
 }
 
-/// Generate an empty struct to be able to define impl blocks (in
-/// `rewrite_impl`) on it for its specification functions.
-pub fn generate_new_struct(item: &syn::ItemImpl) -> syn::Result<syn::ItemStruct> {
-    let name_generator = NameGenerator::new();
-    let struct_name = match name_generator.generate_struct_name(item) {
-        Ok(name) => name,
-        Err(msg) => return Err(syn::Error::new(
-            item.span(),
-            msg,
-        ))
-    };
-    let struct_ident = syn::Ident::new(&struct_name,
-                                       item.span());
-
-    let mut new_struct: syn::ItemStruct = parse_quote_spanned! {item.span()=>
-        #[allow(non_camel_case_types)] struct #struct_ident {}
-    };
-
-    new_struct.generics = item.generics.clone();
-    let generics = &new_struct.generics;
-
-    let mut fields = vec![];
-
-    // Add `PhantomData` markers for each type parameter to silence errors
-    // about unused type parameters.
-    for param in generics.params.iter() {
-        let field: ParsableUnnamedField = match param {
-            syn::GenericParam::Type(tp) => {
-                let ident = tp.ident.clone();
-                syn::parse_quote!{ ::core::marker::PhantomData<#ident> }
+/// Process external specifications in Rust modules marked with the
+/// #[extern_spec] attribute. Nested modules are processed recursively.
+/// Specifications are collected from functions and function stubs.
+///
+/// Modules are rewritten so that their name does not clash with the module
+/// they are specifying.
+pub mod mods {
```

Comment:
Maybe `extern_spec` should be a folder with multiple files in it? It's a bit more readable than having inlined modules in one file.